cockatrice/src/dlg_update.cpp

```diff
@@ -108,40 +103,34 @@ void DlgUpdate::finishedUpdateCheck(bool needToUpdate, bool isCompatible, QVaria
     //If there are no available builds, then they can't auto update.
     enableUpdateButton(isCompatible);
 
-    //If there is an update, save its URL and work out its name
-    if (isCompatible) {
-        QString endUrl = (*build)["path"].toString();
-        updateUrl = API_DOWNLOAD_BASE_URL + endUrl;
-        commitHash = (*build)["sha1"].toString().left(SHORT_SHA1_HASH_LENGTH);
-        commitDate = (*build)["created"].toString().remove(DATE_LENGTH, MAX_DATE_LENGTH);
-    }
-
     //Give the user the appropriate message
-    if (needToUpdate) {
-        if (isCompatible) {
-
-            QMessageBox::StandardButton reply;
-            reply = QMessageBox::question(this, "Update Available",
-                                          "A new build (commit " + commitHash + ") from " + commitDate +
-                                          " is available. Download?",
-                                          QMessageBox::Yes | QMessageBox::No);
-            if (reply == QMessageBox::Yes)
-                downloadUpdate();
-        }
-        else
-        {
-            QMessageBox::information(this, tr("Cockatrice Update"),
-                                     tr("Your version of Cockatrice is out of date, but there are no packages"
-                                                " available for your operating system. You may have to use a developer build or build from source"
-                                                " yourself. Please visit the download page."));
-        }
-    }
-    else {
+    if (!needToUpdate) {
         //If there's no need to update, tell them that. However we still allow them to run the
         //downloader themselves if there's a compatible build
         QMessageBox::information(this, tr("Cockatrice Update"), tr("Your version of Cockatrice is up to date."));
     }
 
+    if (isCompatible) {
+        //If there is an update, save its URL and work out its name
+        updateUrl = release->getDownloadUrl();
+        publishDate = release->getPublishDate().toString(Qt::DefaultLocaleLongDate);
+
+        QMessageBox::StandardButton reply;
+        reply = QMessageBox::question(this, "Update Available",
```

Comment:
Also it seems like tagged unions are a goddamn PITA in cpp: http://en.cppreference.com/w/cpp/language/union

Thoughts on rust? I do have some docs on incorporating rust into a cpp app. I'd be perfectly willing to start extracting all the core bits to safer rust libraries and leave the Qt parts for just UI calling functions exposed from rust FFI.